README.md

```diff
@@ -30,6 +30,107 @@ OPENSHIFT_HOST={openshift_master_uri} SSO_ROUTE={sso_url} yarn start:dev
 example command against PDS cluster
 `OPENSHIFT_HOST=master.CITY.openshiftworkshop.com SSO_ROUTE=secure-sso-sso.apps.CITY.openshiftworkshop.com yarn start:dev`
 
+# Local Development of Walkthroughs
+
+This section describes how to set up the Webapp for Walkthrough authors. It is recommended to put your Walkthroughs in a separate directory that you
+later publish on Github or another git hosting provider.
+
+## Layout of your Walkthroughs directory
+
+The recommended layout for a Walkthroughs directory consists of a subfolder named `walkthrouhgs` and two files:
+
+```
+.
+└── walkthroughs
+    ├── <Walkthrough ID>
+    │   ├── images
+    │   │   └── image.png
+    │   ├── walkthrough.adoc
+    │   └── walkthrough.json
+    ├── 2-walkthrough-two-short-description
+    └── 3-walkthrough-three-short-description
+```
+
+* `Walkthrough ID`: The Walkthrough ID will be used in the URL of the Tutorial Web App. It must be unique and should match `^[a-zA-Z0-9_].$`. 
+* `walkthrough.adoc`: The main content of the Walkthrough in Asciidoc format. It is recommended to have all content in a single file. Please refer to the documentation of [Asciidoctor](https://asciidoctor.org/docs/what-is-asciidoc/) to learn more about Asciidoc.
+* `walkthrough.json`: The Walkthrough manifest file. Currently contains information about Walkthrough extra dependencies.
+
+## walkthrough.json format
+
+The `walkthrough.json` file is used to describe extra dependencies of your walkthrough. At the moment this means either additional service instances or git repositories that your Walkthrough depends on. The basic file structure is:
+
+```json
+{
+  "dependencies": {
+    "repos": [],
+    "serviceInstances": []
+  }
+}
+```
+
+### walkthrough.json: repos
+
+You can specify repositories that will be created for Walkthrough users in the cluster's [Gitea](https://gitea.io/en-us/) instance. The repositories will only be created once a user starts a walkthrough. The format is:
+
+```json
+{
+  "repoName": "<Repo Name in Gitea>",
+  "cloneUrl": "<Optional: clone from external repo>"
+}
+```
+
+### walkthrough.json: serviceInstances
+
+You can also specify additional service dependencies that will allow the users of your Walkthrough to use those services. Please note that the Services have to be present on the cluster already. Adding a service dependency creates
+a service instance that links that service to the user's Walkthrough project.
```

Comment:
It might be worth mentioning here that the web app will expose any route URLs created by the ServiceInstances with the format `route-<route-name>-host`. The value will be in the format `<protocol>://<route>`

Reply:
added below